ts/src/gate.ts

```diff
@@ -2521,42 +2521,54 @@ export default class gate extends Exchange {
         let maxLimit = 1000;
         if (market['contract']) {
             maxLimit = 1999;
-            limit = (limit === undefined) ? maxLimit : Math.min (limit, maxLimit);
             if (market['future']) {
                 method = 'publicDeliveryGetSettleCandlesticks';
             } else if (market['swap']) {
                 method = 'publicFuturesGetSettleCandlesticks';
+            } else if (market['option']) {
+                maxLimit = undefined;
+                method = 'publicOptionsGetCandlesticks';
             }
-            const isMark = (price === 'mark');
-            const isIndex = (price === 'index');
-            if (isMark || isIndex) {
-                request['contract'] = price + '_' + market['id'];
-                params = this.omit (params, 'price');
+            if (!market['option']) {
+                const isMark = (price === 'mark');
+                const isIndex = (price === 'index');
+                if (isMark || isIndex) {
+                    request['contract'] = price + '_' + market['id'];
+                    params = this.omit (params, 'price');
+                }
             }
         }
-        limit = (limit === undefined) ? maxLimit : Math.min (limit, maxLimit);
+        if (market['option']) {
+            limit = undefined;
```

Comment:
@Dan-krm yeah something is wrong on the exchange side, but I think we should try to clarify it with Gate directly

```
handleRestResponse:
 gate GET https://api.gateio.ws/api/v4/options/candlesticks?contract=BTC_USDT-20230609-26500-C&interval=1h&limit=5 200 OK 
ResponseHeaders:
 {
  'Access-Control-Allow-Headers': 'Content-Type,X-Gate-User-Id, DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type',
  'Access-Control-Allow-Methods': 'GET,POST,OPTIONS,DELETE,PUT',
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Max-Age': '1728000',
  'Access-Control-Request-Method': '*',
  Connection: 'keep-alive',
  'Content-Encoding': 'gzip',
  'Content-Type': 'application/json',
  Date: 'Tue, 06 Jun 2023 13:28:54 GMT',
  Server: 'openresty',
  'Transfer-Encoding': 'chunked',
  Vary: 'Origin',
  'X-Ratelimit-Limit': '10'
} 
ResponseBody:
 []
```

Reply:
Sounds good, do you have an open communication channel with gate or should I try to reach out to them?

Reply:
@Dan-krm I did reached out to them, but I don't think it will be fixed any time soon. In that case, can you isolate this logic inside a different method (`fetchOptionOHLCV`) that will be called inside the main `fetchOHLCV`?  to avoid a lot of ifs/elses just to skip parameter passing

Reply:
Yeah sure I'll isolate the logic into `fetchOptionOHLCV`